app/api/server/v1/teams.ts

```diff
@@ -172,7 +172,7 @@ API.v1.addRoute('teams.listRoomsOfUser', { authRequired: true }, {
 
 		const allowPrivateTeam = hasPermission(this.userId, 'view-all-teams', team.roomId);
 
-		if (!hasPermission(this.userId, 'view-all-team-channels', team.roomId)) {
+		if (!hasPermission(this.userId, 'view-all-team-channels', team.roomId) && userId !== this.userId) {
```

Comment:
Hey! Talking with Diego, the change looks good 

Would you mind adding some e2e tests to this endpoint to cover this behavior? 🙏

Reply:
@Darshilp326 Are you planning to continue work on this one?